lib/crypt-port.h

```diff
@@ -179,11 +179,29 @@ _crypt_strcpy_or_abort (void *, const size_t, const void *);
 # define _strong_alias(name, aliasname) \
   extern __typeof (name) aliasname __THROW __attribute__ ((alias (#name)))
 
+/* Starting with GCC 10.2, we can use the symver attribute, which also works
+   with link-time optimization enabled.  */
+# if __GNUC__ > 10 || (__GNUC__ == 10 && __GNUC_MINOR__ >= 2)
+
+/* Referencing specific _compatibility_ symbols still needs inline asm.  */
+# define _symver_ref(extstr, intname, version) \
+  __asm__ (".symver " #intname "," extstr "@" #version)
+
+/* Set the symbol version for EXTNAME, which uses INTNAME as its
+   implementation.  */
+# define symver_set(extstr, intname, version, mode) \
+  extern __typeof (intname) intname __THROW \
+    __attribute__((symver (extstr mode #version)))
```

Comment:
The order of operations here briefly confused me into thinking the `#if` conditional was backward.  Can you please rearrange it like this?

```
#if __GNUC__ > 10 || (__GNUC__ == 10 && __GNUC_MINOR__ >= 2)
/* Starting with GCC 10.2, we can use the symver attribute to define symbol versions.
   This has the advantage of working correctly with link-time optimization.  */

/* Set the symbol version for EXTSTR, which uses INTNAME as its
   implementation. */
# define symver_set(extstr, intname, version, mode) // ...

/* Referencing specific _compatibility_ symbols still needs inline asm.  */
# define _symver_ref(extstr, intname, version) // ...
```

Reply:
Will update this with the next rebase.